.grit/patterns/common.grit

```diff
@@ -145,22 +145,36 @@ predicate AddImport($m, $fromArg) {
 
 predicate AddNamespaceImport($m, $fromArg) {
    $fromArg <: let($alias, $from) and {
-    or {
-      Identifier() as $alias,
-      $_ where $alias = Identifier(name = $m)
+    $_ where or {
+      $m <: Identifier() as $alias,
+      $alias = Identifier(name = $m)
     },
     or {
       StringLiteral() as $from,
       $_ where $from = StringLiteral(value = $fromArg)
     }
   } where {
     $importDeclaration = ImportNamespaceSpecifier(local = $alias, id = null, name = null),
+    ! $importDeclaration <: $alreadySolved,
     $alreadySolved = or { $alreadySolved, $importDeclaration },
     $solvedPairs = or { $solvedPairs, `$importDeclaration, $from` },
     or { and { ! $from <: $froms, $froms = or { $froms, $from } }, true }
   }
 }
 
+predicate ensureNamespaceImportFrom($what, $from) {
+  or {
+    isNamespaceImported($what, $_),
+    AddNamespaceImport($what, $from)
```

Comment:
Syntax here confuses me - are you using the `or` as an `if`?

Also doesn't this need a trailing comma?

Reply:
Yes, `if` would work also, just thought this is more readable considering the first two cases are optional. Let me know if you want this changed.

Reply:
Please use if.

Reply:
Changed.